src/nimodinst/system_tests/test_system_nimodinst.py

```diff
@@ -4,28 +4,29 @@
 import re
 
 
-def test_bad_device_family():
+def test_bad_device_family(family):
     with nimodinst.Session('FAKE') as session:
         assert len(session) == 0
 
 
-def test_no_device_family():
-    with nimodinst.Session('') as session:
+def test_no_device_family(family):
+    with nimodinst.Session(family) as session:
         assert len(session) > 0, 'Must have hardware for ModInst tests to be valid.'
 
 
-def test_device_family_string_with_dashes():
-    with nimodinst.Session('NI-DMM') as session:
-        assert len(session) > 0, 'Must have NI-DMM hardware for this tests to be valid.'
+def test_device_family_string_with_dashes(family):
+    modified_family = family.upper()[:2] + '-' + family.upper()[2:]  # Changing nidmm to NI-DMM
+    with nimodinst.Session(modified_family) as session:
+        assert len(session) > 0, 'Must have hardware for this tests to be valid.'
 
 
-def test_device_family_string_without_dashes():
-    with nimodinst.Session('nidmm') as session:
-        assert len(session) > 0, 'Must have NI-DMM hardware for ModInst tests to be valid.'
+def test_device_family_string_without_dashes(family):
+    with nimodinst.Session(family) as session:
```

Comment:
Similar here, since you can take the family on the command line, make sure there is no dash.